std/os/wasi/core.zig

```diff
@@ -0,0 +1,12 @@
+pub const __wasi_errno_t = u16;
```

Comment:
I think we could probably write this entire file pretty quickly using `zig translate-c` on https://github.com/CraneStation/wasi-sysroot/blob/wasi/libc-bottom-half/headers/public/wasi/core.h and then patching it up.

Let's have the external functions be `extern "wasi"` and that will be a specially understood library known to Zig to mean that the code has depended on a function that is specific to the WASI target. This will help Zig have useful error messages, for example if the user tries to target Windows but then calls a WASI function.

Reply:
That was next on my list. 

Where would the `extern "wasi"` checking take place?

Reply:
You could start with no special treatment. But then probably in the linker we would make sure it doesn't try to do something like `-lwasi`.

Reply:
I have implemented the special treatment. It currently dis-allows `extern "wasi"` on non-wasi platforms. I don't think it disallows `-lwasi`.

The translation will come in a separate PR.